specifyweb/frontend/js_src/lib/components/WorkBench/WbView.tsx

```diff
@@ -37,25 +37,417 @@ import { hasPermission } from '../Permissions/helpers';
+        className="flex items-center justify-between gap-x-1 gap-y-2 whitespace-nowrap"
+        role="toolbar"
+      >
+        <div className="wb-name-container contents">
+          <DataSetName
+            dataset={dataset}
+            getRowCount={() =>
+              hotRef.current?.hotInstance
+                ? dataset.rows.length
+                : hotRef.current.hotInstance.countRows() -
+                  hotRef.current.hotInstance.countEmptyRows(true)
+            }
+          />
+        </div>
+        <Button.Small
+          aria-haspopup="grid"
+          aria-pressed={toolkit.show}
+          className="wb-show-toolkit"
+          onClick={toolkit.toggle}
+        >
+          {commonText.tools()}
+        </Button.Small>
+        <span className="-ml-1 flex-1" />
+        {canUpdate || isMapped ? (
+          <Link.Small href={`/specify/workbench/plan/${dataset.id}/`}>
+            {wbPlanText.dataMapper()}
+          </Link.Small>
+        ) : undefined}
+        <WbActionsComponent
+          dataset={dataset}
+          hasUnSavedChanges={hasUnSavedChanges}
+          isUploaded={isUploaded}
+          triggerRefresh={triggerRefresh}
+          mappings={mappings as WbMapping}
+          checkDeletedFail={checkDeletedFail}
+          spreadSheetUpToDate={spreadSheetUpToDate}
+          workbench={workbench}
+          toggleResults={toggleResults}
+        />
+      </div>
+      {toolkit.show && (
+        <WbToolkit
+          dataset={dataset}
+          hotRef={hotRef}
+          toolkitOptions={toolkitOptions}
+          mappings={mappings as WbMapping}
+          data={data}
+          handleDatasetDelete={handleDatasetDelete}
+          hasUnSavedChanges={hasUnSavedChanges}
+        />
+      )}
+      <div className="flex flex-1 gap-4 overflow-hidden">
+        <section className="wb-spreadsheet flex-1 overflow-hidden overscroll-none">
+          <WbSpreadsheet
+            dataset={dataset}
+            hotRef={hotRef}
+            isUploaded={isUploaded}
+            data={data}
+            validation={workbench.validation!}
+            cells={workbench.cells!}
+            disambiguation={workbench.disambiguation!}
+            hooks={hooks}
+          />
+        </section>
+        {showResults && (
+          <aside aria-live="polite" className="wb-uploaded-view-wrapper">
+            <WbUploaded
+              dataSetId={dataset.id}
+              dataSetName={dataset.name}
+              isUploaded={isUploaded}
+              recordCounts={workbench.validation!.uploadResults.recordCounts}
+              onClose={closeResults}
+            />
+          </aside>
+        )}
+      </div>
+      <WbUtilsComponent
+        isUploaded={isUploaded}
+        searchPreferences={searchPreferences}
+        cellCounts={cellCounts}
+        utils={workbench.utils!}
+        spreadsheetContainer={spreadsheetContainer}
+      />
+    </>
+  );
+}
+
 // REFACTOR: when rewriting to React, add ErrorBoundaries
```

Comment:
What does ErrorBoundaries refer to?

Reply:
Error boundary is a react concept - https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary

If an exception occurs during render inside of the error boundary, the error doesn't crash the rest of the application (the error doesn't spread beyond the error boundary) - this is great for handling crashes a bit more gracefully (crashing only part of the application rather than all at once)

we implement one in here:

https://github.com/specify/specify7/blob/7fa4d8b284570345aa415249bc67aca9b72580d5/specifyweb/frontend/js_src/lib/components/Errors/ErrorBoundary.tsx#L27

use you IDE's "Go to references" (the most essential feature) to find how we use it in the codebase

ie. wrap some components in `<ErrorBoundary>` or `<ErrorBoundary dismissable>` (if the error is likely not critical and so we should let the user dismiss it without page reload - use it when you are reasonably confident that the error won't cause a data corruption/incorrect data being sent to the back-end)

for example, wrap the GeoMap, GeoLocate, CoordinateConversion, AdvancedSearch, Upload Results and other features in this error boundary - these are mostly readonly, so if they crash, we can still let user use the application (i.e at very least, the user would be able to reach the "Save" button so as not to loose their unsaved changes)